src/NewTwitchApi/Resources/StreamsApi.php

```diff
@@ -29,33 +29,26 @@ public function getStreamForUsername(string $username): ResponseInterface
      * @throws GuzzleException
      * @link https://dev.twitch.tv/docs/api/reference/#get-streams
      */
-    public function getStreams(array $userIds = [], array $usernames = [], array $gameIds = [], array $communityIds = [], array $languages = [], int $first = null, string $before = null, string $after = null): ResponseInterface
-    {
-        $queryParamsMap = [];
-        foreach ($userIds as $id) {
-            $queryParamsMap[] = ['key' => 'user_id', 'value' => $id];
-        }
-        foreach ($usernames as $username) {
-            $queryParamsMap[] = ['key' => 'user_login', 'value' => $username];
-        }
-        foreach ($gameIds as $gameId) {
-            $queryParamsMap[] = ['key' => 'game_id', 'value' => $gameId];
-        }
-        foreach ($communityIds as $communityId) {
-            $queryParamsMap[] = ['key' => 'community_id', 'value' => $communityId];
-        }
-        foreach ($languages as $language) {
-            $queryParamsMap[] = ['key' => 'language', 'value' => $language];
-        }
-        if ($first) {
-            $queryParamsMap[] = ['key' => 'first', 'value' => $first];
-        }
-        if ($before) {
-            $queryParamsMap[] = ['key' => 'before', 'value' => $before];
-        }
-        if ($after) {
-            $queryParamsMap[] = ['key' => 'after', 'value' => $after];
-        }
+    public function getStreams(
+        array $userIds = [],
+        array $usernames = [],
+        array $gameIds = [],
+        array $communityIds = [],
+        array $languages = [],
+        int $first = null,
+        string $before = null,
+        string $after = null
+    ): ResponseInterface {
+        $queryParamsMap = [
+            'user_id' => $userIds,
+            'user_login' => $usernames,
+            'game_id' => $gameIds,
+            'community_id' => $communityIds,
+            'language' => $languages,
+            'first' => $first,
```

Comment:
The problem with removing the `if` condition determining if this should be added to the parameter map or not is that, at least with integers, passing in a null value (instead of not including the key/value pair at all) results in the parameter getting added to the query string when it shouldn't. 

As an example, look at this call I made with the CLI client:

```
Which endpoint would you like to call?
0) Quit
1) Validate an Access Token
2) Refresh an Access Token
3) Get an App Access Token
4) Get Games
5) Get Streams
6) Get Users
7) Get User with Access Token
8) Get Users Follows
9) Get Webhook Subscriptions for User
Choice: 5

Get Streams
IDs (separated by commas):
Usernames (separated by commas):
Games (separated by commas): fortnite
Communities (separated by commas):
Languages (separated by commas):
Max results to return:
Cursor value previous page starts with:
Cursor value next page starts with:

/helix/streams?game_id=fortnite&first=0

{
    "error": "Bad Request",
    "status": 400,
    "message": "The parameter \"first\" was malformed: the value must be greater than or equal to 1"
}
```

As you can see, the only value I'm specifiying is the game "Fortnite". The resulting endpoint should therefore be `/helix/streams?game_id=fortnite`, but as you can see it is actually `/helix/streams?game_id=fortnite&first=0`. This is invalid, of course, as the response from Twitch tells us.

That's the reason I had arrays wrapped in `foreach` and other values wrapped in `if` statements: to ensure that *only* the supplied key/values get added to the endpoint/query string.

The new implementation seems to be fine for arrays and strings, so this issue seems limited to integers, where the default `null` value is getting cast to `0` instead of not getting used at all. I haven't tested booleans yet.

Relatedly, I recommend using the CLI client while developing to make sure A) it still works, and B) the endpoints it's calling still work. :-)